extensions/resource-deployment/src/services/tools/toolBase.ts

```diff
@@ -234,14 +235,34 @@ export abstract class ToolBase implements ITool {
 		});
 	}
 
+	/**
+	 * Sets the tool with discovered state and version information.
+	 * Upon error the status field is set to ToolStatus.Error and statusDescription is set to the corresponding error message
+	 * and original error encountered is re-thrown so that it gets bubbled up to the caller.
+	 */
 	public async loadInformation(): Promise<void> {
-		await this._pendingVersionAndStatusUpdate;
+		try {
+			await this._pendingVersionAndStatusUpdate;
+		} catch (error) {
+			this.status = ToolStatus.Error;
+			this._statusDescription = getErrorMessage(error);
+			this._installationPathOrAdditionalInformation = this._statusDescription;
+			throw error;
+		}
 	}
 
-	private startVersionAndStatusUpdate() {
+	/**
+	 * 	Invokes the async method to update version and status for the tool
+	 * 	Any errors from underlying code execution are handled to set this.status and this.statusDescription.
+	 */
+	private startVersionAndStatusUpdate(): void {
+		this._statusDescription = '';
 		this._pendingVersionAndStatusUpdate = this.updateVersionAndStatus();
 	}
 
+	/**
+	 * updates the version and status for the tool. Returns a promise that never rejects but instead handles
```

Comment:
Unfinished comment. Also this isn't necessarily true - what part of this code guarantees it'll never reject? Any error here will cause it to throw (and thus reject)

Reply:
misplaced unfinished comment with respect to latest changes. Fixed.